ansible_collections/arista/avd/roles/eos_designs/doc/network-services.md

```diff
@@ -228,6 +262,23 @@ mac_address_table:
             - < hostname1 >
             - < hostname2 >
 
+        # Explicitly enable or disable l3_multicast to override setting of tenants.<tenant>.l3_multicast.enabled.
+        # Allow override of tenants.<tenant>.l3_multicast.node_settings
+        l3_multicast:
+          enabled: < true | false >
+          node_settings: # Optional
+            # For each group of nodes, allow configuration of EVPN PEG features | Optional
+            - nodes: [ < node_1 >, < node_2 >, < node_N > ]    # Optional
+              evpn_peg:                                        # Optional
+                transit: < true | false | default false >      # Enable EVPN PEG transit mode
+              rp_addresses:                                    # RP Addresses & associated groups
+                - rps:                                         # A minimum of one RP must be specified if node_settings is defined
+                    - < rp_address_1 >
+                    - < rp_address_2 >
+                  groups:                                      # Optional
+                    - < group_prefix_1/mask >
+                    - < group_prefix_2/mask >
+
```

Comment:
Having a deep model of nested lists will make it harder to override only some settings. In your code you are testing for `node_settings` specifically, so everything below will override on VRF level.
I would prefer to split it up a bit, to also open for future support of adjacent features.
```suggestion
        l3_multicast:
          enabled: < true | false >
          evpn_peg:
            # For each group of nodes, allow configuration of EVPN PEG features | Optional
            # At least one RP address must be configured for EVPN PEG to be configured.
            - nodes: [ < node_1 >, < node_2 >, < node_N > ]                # Optional - will apply to all nodes with RP addresses configured if not set.
              transit: < true | false | default false >                    # Enable EVPN PEG transit mode

          rp_addresses:                                                    # For each group of nodes, allow configuration of RP Addresses & associated groups
            - nodes: [ < node_1 >, < node_2 >, < node_N > ]                # Optional - will apply to all nodes if not set.
              rps: [ < rp_address_1 >, < rp_address_2 > ]                  # A minimum of one RP must be specified
              groups: [ < group_prefix_1/mask >, < group_prefix_1/mask > ] # Optional

```

Reply:
Have now implemented this data model.